packages/component-library/stories/Heading.stories.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from "react"
+import Box from "../components/Box"
+import { Heading } from "../components/Heading"
+
+export default { title: "typography/Heading" }
+
+export const Display0 = () => (
+  <Heading variant="display-0">This is a display heading</Heading>
+)
+
+export const Heading1 = () => <Heading variant="heading-1">Heading 1</Heading>
+
+export const Heading2 = () => <Heading variant="heading-2">Heading 2</Heading>
+
+export const Heading3 = () => <Heading variant="heading-3">Heading 3</Heading>
+
+export const Heading4 = () => <Heading variant="heading-4">Heading 4</Heading>
```

Comment:
I really wanted to use levels here (i.e. `<Heading level={1}>` but this wouldn't translate well to the naming in Paragraphs (intro-lede, paragraph-body, as `<Paragraph variant="intro-lede">`, etc). Another alternative for `variant` might be `appearance` (Atlaskit), `type` (Ant Design), `intent` (Redbubble).

Whatever name we use here will be carried through to other typography components and buttons (to start with). 

@katsambarcus @didoesdigital Did you have any thoughts on this one?

Reply:
For what it's worth, my vote is for `variant` across the board.

Reply:
@sebpearce That's looking like it's going to be the most consistent to the rest of the Kaizen components, so i'm inclined to leave it.

Reply:
FWIW: I quite like `appearance` as it's very accurate even if it's a little long. I was thinking `style` which is common in things like Microsoft Word and Google Docs, but that's way overloaded with the meaning in css/stylesheets.

I seem to recall we were on a mission to remove `variant` from all the components because it's too vague, such as renaming `variant` to `mood` in the Notification and Modals.

Reply:
My previous notes on this:

- level
- hierarchy
- variant
- headingSize/paragraphSize
- headingSizing/paragraphSizing
- headingScale/paragraphScale
- scale

My concern with "variant" or "type" is that it is a generic catch all name that could mean basically anything. In a design system, you'd see that used to describe colors, length, "mood", spacing, layout/composition, anything. It's often already taken when you come to it, and means something different across every component. 

That said, "variant" when truly used to mean the main variants of this component, would be the one time I'd be comfortable with it. For Heading, I would expect the "variant" prop to describe the main thing that varies between the main uses of headings: its levels.